spring-integration-core/src/main/java/org/springframework/integration/channel/AbstractMessageChannel.java

```diff
@@ -392,15 +390,17 @@ private boolean sendWithObservation(Message<?> message, long timeout) {
 			Message<?> messageToSendInternal = messageToSend;
 			if (message instanceof ErrorMessage errorMessage) {
 				messageToSendInternal =
-						new ErrorMessage(errorMessage.getPayload(),
+						(errorMessage.getOriginalMessage() != null) ? new ErrorMessage(errorMessage.getPayload(),
```

Comment:
Consider to extract `errorMessage.getOriginalMessage()` into a local variable for some optimization.